### PR TITLE
[Feature:RainbowGrades] Customization UI improvements

### PR DESCRIPTION
**Pull request by Kevin-MM**
This PR builds on #4010 and should be merged after.

### Please check if the PR fulfills these requirements:

* [x] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)
* [x] Tests for the changes have been added/updated (if possible)
* [x] Documentation has been updated/added if relevant

### What is the current behavior?
Previously the Rainbow Grades Customization UI didn't give any useful feedback about when the server-side processes had completed.

### What is the new behavior?
The UI form now displays a few different status messages indicating the status of rainbow grades generation, specifically when the generation has completed.

See #4010 for instructions on how to use.

A few small bug fixes and visual improvements are included.

### Other information?
Tested manually inside 'sample course'

### CODE REVIEW

File: migration/migrator/migrations/course/20190628100043_create_rainbow_grades_dir.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import os
+import grp
+from pathlib import Path
```
Comment (MasterOdin) [inline migration/migrator/migrations/course/20190628100043_create_rainbow_grades_dir.py]: These migration files should probably be combined.

Comment (bmcutler) [reply migration/migrator/migrations/course/20190628100043_create_rainbow_grades_dir.py]: this PR will be merged after another PR which introduces these migrations.  
But yes, I agree, the migrations should be combined (and re-numbered to be 201907XX).
And then removed from this PR.

**Comment by Kevin-MM** [comment]
Removed two old migrators that had been merged into a single one.  Also made a few small UI improvements.

**Comment by bmcutler** [comment]
If I get in a state where rainbowgrades errors (don't worry about debugging the rainbowgrade error right now)  

But the page gets stuck with status saving or with "Status: Generating rainbow grades, please wait..."
for a *long* time.  Eventually it says a failure occured.  The auto_debug_output.txt is already updated with the error.  What additional timeout are we waiting for?

root@vagrant:/var/local/submitty/courses/s19/development/rainbow_grades# more auto_debug_output.txt 
rsync: failed to set times on "/var/local/submitty/courses/s19/development/rainbow_grades/raw_data/.": Operation not permitted (1)
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1196) [sender=3.1.2]
make: *** [pull_test] Error 23
process_grades.out: /usr/local/submitty/GIT_CHECKOUT/RainbowGrades/output.cpp:602: void start_table_output(bool, const std::vector<Student*>&, int, int, i
nt, int, Student*, Student*, Student*, Student*, Student*): Assertion `GRADEABLES[g].getPercent() < 0.01' failed.
make: *** [overall] Aborted (core dumped)
rsync: link_stat "/var/local/submitty/courses/s19/development/rainbow_grades/individual_summary_html/*" failed: No such file or directory (2)
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1196) [sender=3.1.2]
make: *** [push_test] Error 23

**Comment by bmcutler** [comment]
I'm confused what is happening.  There are no jobs in the queue, the new error log output file is produced, but the page is still hanging for 60 seconds.

**Comment by Kevin-MM** [comment]
I believe this problem is occurring because I am waiting for output.html to be produced in rainbow_grades directory to signal when the c++ part of the code has finished running.  In this case the file doesn't get produced so the script times out.  Right now the default timeout is set at 60 seconds, but I think we can probably shorten it.  I'm not sure this bit of code is required as I think I put it in there before I added some other checks so let me remove it and do some testing.

**Comment by bmcutler** [comment]
Now it just always says "successfully generated" even when it failed.
So there's a special case check for the category "Note", which is not supposed to count for a grade.
(We may indeed want to change this, but for now it's a convenient way to check failure).

To reproduce, drag the "Note" category into the assigned buckets, and give it a percentage value > 0.
Then you should get this error:
Assertion `GRADEABLES[g].getPercent() < 0.01'
in auto_debug_output.txt

And output.html is not generated.

And the individual_summary_html is not generated (and wont be copied to ../reports/summary_html)
But if you had an earlier successful run, the old files will still be there.

Regardless, it always says "Rainbow grades successfully generated!" when the job completes, even if there was an error and (new) files were not produced.


Also, it failed travis after the last commit, so I don't want to merge until I see the green check again :) 
It might be the need to merge master apache config thing.  If it passes in the morning (even if we don't have this status this sorted), I'll merge.

File: .setup/apache/submitty.conf [hunk 2]
```diff
@@ -1,144 +0,0 @@
-# If hosting more than just Submitty on Apache, you should set the
```
Comment (bmcutler) [inline .setup/apache/submitty.conf]: oops, I don't think your PR should be deleting this file.  That is likely why its failing the travis e2e tests

Comment (Kevin-MM) [reply .setup/apache/submitty.conf]: Readded this file.

**Comment by Kevin-MM** [comment]
Updated code so a failure of the rainbow grades c++ code no longer produces "successfully generated" in the ui.

On another note I did notice that setting the Note category to 1% does not produce this error.

**Pull request merged**